doc/design/garbage-collection-uploads-asset-blobs.md

```diff
@@ -0,0 +1,26 @@
+# Upload and Asset Blob Garbage Collection
+
+## Background
+
+Now that the [design for S3 trailing delete](https://github.com/dandi/dandi-archive/blob/master/doc/design/s3-trailing-delete.md) is deployed to staging, we are ready to implement garbage collection. [This older design document](https://github.com/dandi/dandi-archive/blob/master/doc/design/garbage-collection-1.md#uploads) is still relevant, and summarizes the various types of garbage collection we want to implement. This document will present a design for garbage collection of uploads and asset blobs, i.e. garbage that accumulates due to improper uploads done by users. A design for garbage collection of orphaned “Assets” (i.e. files that have been properly uploaded, have metadata, etc. but are no associated with any dandisets) is more complex and is left for a future design document.
```

Comment:
i think it would be good to add this type of asset below under the next section. in the dandi model this would be the biggest source of garbage as we mint a new asset each time an asset or blob is modified. 

on this note, can we run a quick analysis of the numbers in each category? i.e. does skipping this for a future iteration really address the issue.

Reply:
> i think it would be good to add this type of asset below under the next section. in the dandi model this would be the biggest source of garbage as we mint a new asset each time an asset or blob is modified.

The intention of this document is to only cover asset blob/upload garbage collection, since these are more straightforward to implement than `Asset` garbage collection (with most of the complexity there surrounding (1) are there cases where we want to retain old assets in the DB, and (2) the fundamental incompatibility the current `Asset` data model has with garbage collection due to the `Asset.previous` pointer). 

> on this note, can we run a quick analysis of the numbers in each category? i.e. does skipping this for a future iteration really address the issue.

Yes, I can get these numbers and add them here.

Reply:
I added these numbers as of today to the design doc (https://github.com/dandi/dandi-archive/pull/1733/commits/5f1301a478a471e818a916cee29bda639002b5bb).